search/tests/test_search_result_processor.py

```diff
@@ -0,0 +1,338 @@
+        test_string_compare = SearchResultProcessor.decorate_matches(test_string, u"الاستحسان")
+        excerpt = srp.excerpt
+        self.assertNotEqual(excerpt, test_string_compare)
+        self.assertTrue(u"note about <b>الاستحسان</b> and it is" in excerpt)
+
+        test_string = (
+            u"Here is a note about stuff and it is very long - more than the desirable length of 100"
+            u" characters - indeed this should show up but it should trim the الاستحسان characters around in"
+            u" order to show the selected text in bold"
+        )
+        test_result = {
+            "content": {
+                "notes": test_string,
+            }
+        }
+        srp = SearchResultProcessor(test_result, u"الاستحسان")
+        test_string_compare = SearchResultProcessor.decorate_matches(test_string, u"الاستحسان")
+        excerpt = srp.excerpt
+        self.assertNotEqual(excerpt, test_string_compare)
+        self.assertTrue(u"should trim the <b>الاستحسان</b> characters around" in excerpt)
+
+    def test_excerpt_front(self):
+        """ test that we process correctly when match is at the front of the excerpt """
+        test_result = {
+            "content": {
+                "notes": "Dog - match upon first word",
+            }
+        }
+        srp = SearchResultProcessor(test_result, "dog")
+        self.assertEqual(srp.excerpt, "<b>Dog</b> - match upon first word")
+
+        test_result = {
+            "content": {
+                "notes": (
+                    "Dog - match upon first word "
+                    "The long and winding road "
+                    "That leads to your door "
+                    "Will never disappear "
+                    "I've seen that road before "
+                    "It always leads me here "
+                    "Lead me to you door "
+                    "The wild and windy night "
+                    "That the rain washed away "
+                    "Has left a pool of tears "
+                    "Crying for the day "
+                    "Why leave me standing here "
+                    "Let me know the way "
+                    "Many times I've been alone "
+                    "And many times I've cried "
+                    "Any way you'll never know "
+                    "The many ways I've tried "
+                    "But still they lead me back "
+                    "To the long winding road "
+                    "You left me standing here "
+                    "A long long time ago "
+                    "Don't leave me waiting here "
+                    "Lead me to your door "
+                    "But still they lead me back "
+                    "To the long winding road "
+                    "You left me standing here "
+                    "A long long time ago "
+                    "Don't leave me waiting here "
+                    "Lead me to your door "
+                    "Yeah, yeah, yeah, yeah "
```

Comment:
@e-kolpakov Are these lyrics under a compatible license? I know it may seem silly, but it's something we probably need to be sure of.